xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiServletURLFactory.java

```diff
@@ -607,12 +608,60 @@ public URL createAttachmentURL(String filename, String spaces, String name, Stri
                         querystring, xwikidb, context);
                 }
             } catch (XWikiException e) {
-                if (LOGGER.isErrorEnabled()) {
-                    LOGGER.error("Exception while trying to get attachment version !", e);
+                LOGGER.error("Exception while trying to get attachment version !", e);
+            }
+        }
+
+        // In the context of the current doc, no need to reload it.
```

Comment:
Why do you need to reload anything? Maybe this needs to be explained in the source.

Reply:
Seems to me "load" would make more sense than "reload".